src/adc.rs

```diff
@@ -46,9 +58,10 @@ pub struct Adc {
     sample_time: AdcSampleTime,
     align: AdcAlign,
     precision: AdcPrecision,
+    stored_cfg: Option<(AdcSampleTime, AdcAlign, AdcPrecision)>,
```

Comment:
Couldn't you just store this on the stack for the functions, e.g. the stash (should then be renamed) just gives the tuple back and restore just takes it. There's no real need to always allocate space for this (i think)

Reply:
This could be done, I suspect the storage use is incredibly minimal but I can correct.

Reply:
It would be saving 3 bytes on a structure that can only ever be instantiated once.

Reply:
Done